gtsam/discrete/DiscreteConditional.cpp

```diff
@@ -45,7 +45,8 @@ template class GTSAM_EXPORT
 /* ************************************************************************** */
 DiscreteConditional::DiscreteConditional(const size_t nrFrontals,
                                          const DecisionTreeFactor& f)
-    : BaseFactor(f / (*f.sum(nrFrontals))), BaseConditional(nrFrontals) {}
+    : BaseFactor(f / f.sum(nrFrontals)->toDecisionTreeFactor()),
```

Comment:
Is that a no-op if already a DecisionTreeFactor?

Reply:
It's not actually. In the simple case where there are no parents, we'll get a `DecisionTreeFactor` with a single value as a `DiscreteFactor::shared_ptr`. We can cast it explicitly, but I used `toDecisionTreeFactor` to do it instead as a shortcut.